### PR TITLE
Support expint rewrite as integral

### LINKED ISSUE #26134: expint: support rewrite as integral
The following doesn't work.  Would be nice if it did!

```
>>> f = expint(n,x)
>>> f.rewrite(Integral)
       ∞
       ⌠
       ⎮  -n  -t⋅x
       ⎮ t  ⋅ℯ     dt
       ⌡
       1
```
**update**: ~~Similarly should work for `Si` and `Ci` too.~~  *Filed #26144 for this*.

**update**: ~~maybe also `polylog`~~ that should be a new issue.

### Notes for potential contributors:

1. Functions such as `elliptic_e` can already do this:
```
>>> f = elliptic_e(phi, n)
>>> f.rewrite(Integral)
φ                        
⌠                        
⎮    _________________   
⎮   ╱        2           
⎮ ╲╱  - n⋅sin (t) + 1  dt
⌡                        
0                        
```
2. Read the help of `f.rewrite?`.
3. Look at `f._eval_rewrite??`
4. Look at `f._eval_rewrite_as_Integral??`

### PR DESCRIPTION
**Pull request by Ravindu-Hirimuthugoda**
#### References to other Issues or PRs

Fixes: https://github.com/sympy/sympy/issues/26134

#### Brief description of what is fixed or changed
Support rewrite as integral for expint

#### Release Notes

* functions
  * support expint rewrite as integral.

### CODE REVIEW

File: sympy/functions/special/error_functions.py [hunk 1]
```diff
@@ -1,15 +1,15 @@
 """ This module contains various functions that are special cases
     of incomplete gamma functions. It should probably be renamed. """
 
-from sympy.core import EulerGamma  # Must be imported from core, not core.numbers
```
Comment (cbm755) [inline sympy/functions/special/error_functions.py:4]: I don't know if the existing comment applies to `oo` or not.  Perhaps better to do the `oo` import on a new line?

Comment (oscarbenjamin) [reply sympy/functions/special/error_functions.py:4]: It would be better to just use `S.EulerGamma`.

Comment (oscarbenjamin) [reply sympy/functions/special/error_functions.py:4]: And `S.Infinity`.

File: sympy/functions/special/error_functions.py [hunk 2]
```diff
@@ -1433,6 +1433,12 @@ def _eval_aseries(self, n, args0, x, logx):
 
         return super(expint, self)._eval_aseries(n, args0, x, logx)
 
+    def _eval_rewrite_as_Integral(self, *args, **kwargs):
+        from sympy.integrals.integrals import Integral
+        n, x = self.args
+        t = Dummy('t')
+        return Integral((t**-n) * (exp(-t*x)), (t, 1, S.Infinity))
```
Comment (sylee957) [inline sympy/functions/special/error_functions.py:1440]:
```suggestion
        return Integral(t**-n * exp(-t*x), (t, 1, S.Infinity))
```

I think that you can remove some unnecessary parenthesis

File: sympy/functions/special/tests/test_error_functions.py [hunk 3]
```diff
@@ -814,3 +815,7 @@ def test_fresnel_series():
     assert (fresnelc(1/z).series(z) - fc.subs(z, 1/z)).expand().is_Order
     assert ((2*fresnels(3*z)).series(z, oo) - 2*fs.subs(z, 3*z)).expand().is_Order
     assert ((3*fresnelc(2*z)).series(z, oo) - 3*fc.subs(z, 2*z)).expand().is_Order
+
+
+def test_issue_26134():
+    assert expint(n, x).rewrite(Integral).dummy_eq(Integral((t**-n) * (exp(-t*x)), (t, 1, oo)))
```
Comment (sylee957) [inline sympy/functions/special/tests/test_error_functions.py:821]:
```suggestion
    assert expint(n, x).rewrite(Integral).dummy_eq(Integral(t**-n * exp(-t*x), (t, 1, oo)))
```

**Comment by cbm755** [comment]
LGTM!

**Pull request merged**